iree/compiler/Conversion/LinalgToSPIRV/VectorizeMemref.cpp

```diff
@@ -360,6 +364,38 @@ class ProcessPlaceHolder final
   }
 };
 
+// Folds linalg.reshape op that directly reshaping an iree.placeholder op into
+// the iree.placeholder op itself.
+class FoldReshapeIntoPlaceholder final
```

Comment:
It seems odd that this is here. Reshapes should be folded with the `hal.interface.*` operations after/during HLOToLinalgOnTensors or after HLOToLinalgOnBuffers. Is there an extra reshape inserted later than that?

Reply:
This is generated when we convert depthwise mhlo.conv into linalg.depthwise_conv op in HLOToLinalgOnBuffers pass. I don't see we have a pattern there to fold linalg.reshape into iree.placeholder?

Reply:
Just add this pattern there?

Reply:
SG. Moved.